### ISSUE TITLE
`lineWrappingIndentation` falsely detects incorrect indentation for text blocks

### ISSUE THREAD

**Issue by xpple** [checkstyle/checkstyle#15939]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac A.java

/var/tmp $ cat checkstyle.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">

    <property name="charset" value="UTF-8"/>

    <property name="severity" value="${org.checkstyle.google.severity}" default="warning"/>

    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="SuppressWarningsFilter"/>

    <module name="TreeWalker">
        <module name="SuppressWarningsHolder"/>

        <module name="Indentation">
            <property name="lineWrappingIndentation" value="4"/>
            <property name="forceStrictCondition" value="true"/>
        </module>
    </module>
</module>

/var/tmp $ cat A.java
public class A {
    private static final String EXAMPLE = """
        Example string""";
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.20.1-all.jar -c checkstyle.xml A.java
Starting audit...
[WARN] A.java:3:23: '"""' has incorrect indentation level 22, expected level should be 8. [Indentation]
Audit done.
```

For Windows users, please use `type` instead of `cat` and run

```cmd
set RUN_LOCALE="-Duser.language=en -Duser.country=US"
java %RUN_LOCALE% -jar checkstyle-10.20.1-all.jar -c checkstyle.xml A.java
```

in place of the last 2 commands above.

---

**Describe what you expect in detail.**
The indentation is correct, so no warning should be reported. Note that the warning can be suppressed by adding `@SuppressWarnings("checkstyle:indentation")` to the declaration, given that `SuppressWarningsFilter` is configured.

It seems to me that Checkstyle wants the ending `"""` on a new line. This is not a notational request, though; it'd be a different string as an extra new line character would be added.

See [JEP 378](https://openjdk.org/jeps/378) for more information on text blocks.

**Comment by romani** [comment]
Check should completely skip indentation validation for all nodes under `TEXT_BLOCK_LITERAL_BEGIN`

**Comment by mohitsatr** [comment]
hi @romani ,I'm trying to debug the input.

However I'm getting language level error :

![Screenshot from 2024-12-22 09-58-55](https://github.com/user-attachments/assets/f4d1fb4a-5f61-4ffa-b455-23d4b777c619)

and it does not go away even after I have set level to 21 

![Screenshot from 2024-12-22 09-53-41](https://github.com/user-attachments/assets/9295c61a-5ec8-4be9-966b-10a26adfefa5)

**Comment by mohitsatr** [comment]
also,
`//` for verification becomes the part of the block-Text/String. see the below image :
 
![Screenshot from 2024-12-22 10-24-20](https://github.com/user-attachments/assets/ed263302-d9d2-4669-9bb9-3385f458395f)

is there a workaround to avoid this ?
thanks in advance.

**Comment by romani** [comment]
Idea should be using jdk11 
All inputs should be out of Ideal compilation as all of them are for different jdk, see comment at first line of Input file.

Just exclude Input from sources to compile.

**Comment by mohitsatr** [comment]
ok like you said, I made it skip `TEXT_BLOCK_LITERAL_BEGIN` node and it's children and logic is working. error is gone.

**Comment by xpple** [comment]
Just FYI, ignoring the check for `TEXT_BLOCK_LITERAL_BEGIN` nodes is not a _fix_ to this issue. If you set `lineWrappingIndentation` to four, the following would be correct:
```java
public class A {
    private static final String EXAMPLE = """
        Example string""";
}
```
The following would not:
```java
public class A {
    private static final String EXAMPLE = """
            Example string"""; // violation
}
```
The first nonwhitespace character of a text block determines the offset of the text block; in both cases the resulting string is the same.

**Comment by romani** [comment]
@mohitsatr, please make PR.

**Comment by romani** [comment]
@xpple , do you see my point that this Check should avoid text blocks at all?

**Comment by xpple** [comment]
I see now that this issue was closed, but I forgot to respond.

> [@xpple](https://github.com/xpple) , do you see my point that this Check should avoid text blocks at all?

No, I don't. Notice that the following strings are equal:
```java
private static final String A = """
    a
    """;

private static final String B = """
        a
        """;
```
That is, `A.equals(B)`. However, the indentation is different. In the first string, the indentation is `4`, and in the second string the indentation is `8`. When the user has defined `lineWrappingIndentation` at `4`, only the first example should pass the check.
